pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -260,17 +287,34 @@ registry = "$registry"
 replace-with = "vendored-sources"
 EOF
     done
+    mkdir $out/registry
 
     for crate in ${toString depCrates}; do
       # Link the crate directory, removing the output path hash from the destination.
-      ln -s "$crate" $out/$(basename "$crate" | cut -c 34-)
-
+      # When the crate directory has a directory directive, putting it to git-* directory.
       if [ -e "$crate/.cargo-config" ]; then
         key=$(sed 's/\[source\."\(.*\)"\]/\1/; t; d' < "$crate/.cargo-config")
+        directory=$(sed 's/directory = "\(.*\)"/\1/; t; d' < "$crate/.cargo-config")
+        package_name=$(sed 's/#pkg: \(.*\)/\1/; t; d' < "$crate/.cargo-config")
+        if [[ ! -z "$directory" ]]; then
+          gitdir=$(basename "$directory")
+          if [ ! -d $out/$gitdir ] ; then
+            mkdir $out/$gitdir
+          fi
+          ln -s "$crate" $out/$gitdir/$(basename "$crate" | cut -c 34-)
+          # This is to handle the case of referencing a package with a relative path.
+          if [ ! -e "$out/$gitdir/$package_name" ] ; then
+            ln -s "$crate" "$out/$gitdir/$package_name"
+          fi
+        else
+          ln -s "$crate" $out/$(basename "$crate" | cut -c 34-)
```

Comment:
When is this branch reached?

Reply:
When there are multiple packages or references in the same git.

Reply:
It will never be reached, so it can be deleted.